### PR TITLE
Added idp hinting in RCauth plugin. Fixed bug in function params

### PR DESCRIPTION
**Pull request by ioigoume**
- Idp hinting for RCAuth plugin
- fixed serious bug

**We need to add a new column in the database**
`alter table cm_rcauth_sources add column idphint varchar(256);`

### CODE REVIEW

**Comment by ioigoume** [comment]
I successfully tested the feature against rciam/rciam-3.1.x branch. After testing i made a few more changes like:
- better logging
- corrected parameter functions of inherited function
- added check for refresh token, since it is not always available

File: local/Plugin/RcauthSource/Lib/lang.php [hunk 1]
```diff
@@ -51,5 +51,7 @@
 	'pl.rcauthsource.secret'         => 'Client Secret',
 	'pl.rcauthsource.secret.desc'    => 'Client Secret obtained from registering with the RCAUTH Public ID',
 	'pl.rcauthsource.issuer'         => 'DN Issuer',
-	'pl.rcauthsource.issuer.desc'    => 'The issuer of the DN certificate'
+	'pl.rcauthsource.issuer.desc'    => 'The issuer of the DN certificate',
```
Comment (NicolasLiampotis) [inline local/Plugin/RcauthSource/Lib/lang.php]: The DN of the certificate issuer

File: local/Plugin/RcauthSource/Model/RcauthSource.php [hunk 2]
```diff
@@ -53,6 +53,11 @@ class RcauthSource extends AppModel {
 			'allowEmpty' => true,
 			'message' => 'Currently only Bearer token type is supported.'
 		),
+		'idphint' => array(
+			'rule' => 'notBlank',
```
Comment (NicolasLiampotis) [inline local/Plugin/RcauthSource/Model/RcauthSource.php]: If the idp hint is supposed to be a URL change this to `'rule' => array('url', true),`

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: local/Plugin/RcauthSource/Model/RcauthSource.php (hunk 2 above)
Comment (ioigoume) [reply local/Plugin/RcauthSource/Model/RcauthSource.php]: Done

File: local/Plugin/RcauthSource/Lib/lang.php (hunk 1 above)
Comment (ioigoume) [reply local/Plugin/RcauthSource/Lib/lang.php]: Done

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**